x-pack/test/cases_api_integration/security_and_spaces/tests/trial/internal/get_connectors.ts

```diff
@@ -232,6 +232,92 @@ export default ({ getService }: FtrProviderContext): void => {
     });
 
     describe('push', () => {
+      describe('externalService', () => {
+        it('sets externalService field to the most recent push info', async () => {
+          const { postedCase, connector } = await createCaseWithConnector({
+            supertest,
+            serviceNowSimulatorURL,
+            actionsRemover,
+          });
+
+          await pushCase({
+            supertest,
+            caseId: postedCase.id,
+            connectorId: connector.id,
+          });
+
+          const pachedCase = await createComment({
+            supertest,
+            caseId: postedCase.id,
+            params: postCommentUserReq,
+          });
+
+          const serviceNow2 = await createConnector({
+            supertest,
+            req: {
+              ...getServiceNowConnector(),
+              name: 'ServiceNow 2 Connector',
+              config: { apiUrl: serviceNowSimulatorURL },
+            },
+          });
+
+          actionsRemover.add('default', serviceNow2.id, 'action', 'actions');
+
+          // change to serviceNow2 connector
+          await updateCase({
+            supertest,
+            params: {
+              cases: [
+                {
+                  id: pachedCase.id,
+                  version: pachedCase.version,
+                  connector: {
+                    id: serviceNow2.id,
+                    name: 'ServiceNow 2 Connector',
+                    type: ConnectorTypes.serviceNowITSM,
+                    fields: {
+                      urgency: '2',
+                      impact: '2',
+                      severity: '2',
+                      category: 'software',
+                      subcategory: 'os',
+                    },
+                  },
+                },
+              ],
+            },
+          });
+
+          await pushCase({
+            supertest,
+            caseId: pachedCase.id,
+            connectorId: serviceNow2.id,
+          });
+
+          const [userActions, connectors] = await Promise.all([
+            getCaseUserActions({ supertest, caseID: postedCase.id }),
+            getConnectors({ caseId: postedCase.id, supertest }),
+          ]);
+
+          const pushes = userActions.filter((ua) => ua.type === ActionTypes.pushed);
+          const latestPush = pushes[pushes.length - 1];
+
+          expect(Object.keys(connectors).length).to.be(2);
+          expect(connectors[serviceNow2.id].push.latestUserActionPushDate).to.eql(
+            latestPush.created_at
+          );
+          expect(connectors[serviceNow2.id].push.externalService?.connector_id).to.eql(
+            serviceNow2.id
+          );
+          expect(connectors[serviceNow2.id].push.externalService?.connector_name).to.eql(
+            serviceNow2.name
+          );
+          expect(connectors[serviceNow2.id].push.externalService?.connector_name).to.not.eql(
```

Comment:
might as well also add above 

`expect(connectors[serviceNow2.id].push.externalService?.connector_id).to.not.eql(connector.id);`